woocommerce/woocommerce-android
# Using site time zone on Analytics

Part of: #9887


## Why
As clarified in https://github.com/woocommerce/woocommerce-android/issues/9887#issuecomment-1855990495, we'd like to use the store time zone to calculate the latest time range like today/this week/this month/this year for scenarios when the store time zone and device time zone are on different dates. More context  in pe5pgL-4bG-p2 

### Description
This PR fixes the issue by updating the range selection functions and using dates on the store's time zone.

### Testing instructions
To test these changes, you must set your device to a different time zone than your store. I used a store in the US and set the device time zone to Australia. This will create a time zone difference of one day (Thursday in the store should be Friday on the device)

1. Open the store and place a new order (on the web)
2. Open the app and check that Today's stats are updated accordingly using the store's timezone.

### Images/gif

|  /  | Before | After |
| ------------- | ------------- |  ------------- |
| My Store  | <img width="300" src="https://github.com/woocommerce/woocommerce-android/assets/18119390/5f6d1868-3e60-4c0d-bba5-e91b9ec15fa6" />  | <img width="300" src="https://github.com/woocommerce/woocommerce-android/assets/18119390/2db63c45-7934-4313-a226-f95728540cda" />l  |
| Analytic Hub  | <img width="300" src="https://github.com/woocommerce/woocommerce-android/assets/18119390/ea9dd7a2-ff1c-4cdd-bdd4-20c8da863724" />  | <img width="300" src="https://github.com/woocommerce/woocommerce-android/assets/18119390/e107a673-d119-4e20-a55c-529efaa31a5c" />  |
